### DISCUSSION TITLE
Can't upgrade argo-workflows due to CRD upgrade

### DISCUSSION THREAD

**Discussion by rconjaerts** [argoproj/argo-helm#1833] [Q&A] [answered]
Hi all,

I'm trying to upgrade argo-workflows to v3.4.5 through these helm charts, but I run into an error due to CRDs. The complete error is:

> UPGRADE FAILED: rendered manifests contain a resource that already exists. Unable to continue with update: CustomResourceDefinition "clusterworkflowtemplates.argoproj.io" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "argo"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "infra"

The command I'm running to upgrade it is pretty basic:
`helm upgrade argo argo/argo-workflows -f values.yaml -n infra --version 0.22.10`

If I put the `install` flag to `false` in case of `crds` it does upgrade, but I would prefer that the crds are upgraded too. I've tried adding the requested annotations but that didn't seem to help.

Does anyone know what I'm doing wrong?

**Comment by yu-croco** [comment, marked as answer]
Hi @rconjaerts , which Helm Chart version did you use originally? 👀 

CRD is moved to `argo-workflows/templates/crds` since `v0.19.0` .  PR: https://github.com/argoproj/argo-helm/pull/1472
If you use `v0.18.0` or earlier and want to upgrade to latest version (`0.22.10`), I think [this comment](https://github.com/argoproj/argo-helm/pull/1472#issuecomment-1342033780) helps you.

**Reply by rconjaerts** [reply]
Hi @yu-croco, we used version 0.11.20 🙈 We haven't updated argo in a very long time.

I ran the code in the comment you shared, after setting the proper namespace value, and this resolved the issue. Thank you very much!

**Reply by yu-croco** [reply]
Thank you for checking, @rconjaerts !
Oh... 🙈 but glad it worked! :)

**Reply by rconjaerts** [reply]
1 small question though, I got the same issue for argo-events. How did you know that all those files needed to be changed to make it possible to update the crd's? I tried looking which files I needed to change for argo-events, but didn't get further then eventbus.argoproj.io, eventsources.argoproj.io, and workfloweventbindings.argoproj.io, but I think those aren't all the files that need updating.

**Reply by yu-croco** [reply]
Basically we (argo-helm maintainers) follow upstreams crd folder, like below. 🙋 

- argo cd https://github.com/argoproj/argo-cd/tree/master/manifests/crds
- argo workflows https://github.com/argoproj/argo-workflows/tree/master/manifests/base/crds
- argo events https://github.com/argoproj/argo-events/tree/master/manifests/base/crds

**Reply by rconjaerts** [reply]
Amazing, thank you very much. Have a great weekend!
